### PR TITLE
chore(`common::shell`): finish implementation + enforce  in `clippy`

### PR DESCRIPTION
**Pull request by zerosnacks**
## Motivation

Enforces `common::shell` now in `clippy` and adds the final implementation of it

## Solution

We still allow the use of `print` without the common shell in tests. In some cases you may need to add a `#[allow(clippy::disallowed_macros)]` if not using the test macros.

Some minor inlining was required / preferred to work around a circular dependency in `fmt`

Note: there are still some places that use `Yansi::paint`, I think these generally make sense contextually (for example to visually differentiate between success / pending / failure in test results).

### CODE REVIEW

File: crates/script/src/verify.rs [hunk 1]
```diff
@@ -263,7 +258,9 @@ fn check_unverified(
                 .unwrap_or_default();
 
             if &current_commit != commit {
-                println!("\tScript was broadcasted on commit `{commit}`, but we are at `{current_commit}`.");
+                let _ = sh_warn!(
+                    "Script was broadcasted on commit `{commit}`, but we are at `{current_commit}`."
```
Comment (zerosnacks) [inline crates/script/src/verify.rs:262]: Does not use \t as formatting looks strange w/ prefixed `Warning: `

File: crates/config/src/lib.rs [hunk 2]
```diff
@@ -957,12 +957,7 @@ impl Config {
             return match rx.recv_timeout(Duration::from_secs(1)) {
                 Ok(res) => res,
                 Err(RecvTimeoutError::Timeout) => {
-                    eprintln!(
-                        "{}",
-                        yansi::Paint::yellow(
-                            "Pulling Docker image for eof-solc, this might take some time..."
-                        )
-                    );
+                    // sh_warn!("Pulling Docker image for eof-solc, this might take some time...");
```
Comment (grandizzy) [inline crates/config/src/lib.rs:960]: any reason to comment this message, or just a leftover?

Comment (zerosnacks) [reply crates/config/src/lib.rs:960]: Ah.. had missed re-enabling this, there is a circular dependency here that has to be fixed.

Comment (zerosnacks) [reply crates/config/src/lib.rs]: Fixed this by allowing `eprintln!` to occur here given it is outside of a common execution path and there is no simple fix for the circular dependency without a major refactor

File: crates/common/fmt/src/dynamic.rs [hunk 3]
```diff
@@ -132,18 +132,6 @@ pub fn format_tokens_raw(tokens: &[DynSolValue]) -> impl Iterator<Item = String>
     tokens.iter().map(format_token_raw)
 }
 
-/// Prints slice of tokens using [`format_tokens`] or [`format_tokens_raw`] depending on `json`
```
Comment (grandizzy) [inline crates/common/fmt/src/dynamic.rs:135]: can we keep the helper fn to print tokens and avoid duplicated code above?

**Review by grandizzy — COMMENTED** [review]
Looks good! Have one nit re duplicated code

File: crates/common/fmt/src/dynamic.rs (hunk 3 above)
Comment (zerosnacks) [reply crates/common/fmt/src/dynamic.rs:135]: Done, had to move the helper into the call location due to a circular dependency in `fmt` / `common`

**Review by grandizzy — APPROVED** [review]
lgtm!

File: clippy.toml [hunk 4]
```diff
@@ -3,10 +3,10 @@ msrv = "1.80"
...
-#     { path = "std::eprint", reason = "use `sh_eprint` or similar macros instead" },
-#     { path = "std::println", reason = "use `sh_println` or similar macros instead" },
-#     { path = "std::eprintln", reason = "use `sh_eprintln` or similar macros instead" },
-# ]
+disallowed-macros = [
+    # See `foundry_common::shell`
+    { path = "std::print", reason = "use `sh_print` or similar macros instead" },
+    { path = "std::eprint", reason = "use `sh_eprint` or similar macros instead" },
+    { path = "std::println", reason = "use `sh_println` or similar macros instead" },
+    { path = "std::eprintln", reason = "use `sh_eprintln` or similar macros instead" },
```
Comment (mattsse) [inline clippy.toml:11]: til

**Review by mattsse — APPROVED** [review]

**Pull request merged**